src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/ApprenticeshipIncentiveDataRepository.cs

```diff
@@ -51,6 +52,21 @@ public async Task<ApprenticeshipIncentiveModel> FindByApprenticeshipId(Guid ince
             return null;
         }
 
+        public async Task<ApprenticeshipIncentiveModel> FindApprenticeshipIncentiveByUlnWithinAccountLegalEntity(long uln, long accountLegalEntityId)
+        {
+            var apprenticeships = await _dbContext.ApprenticeshipIncentives.Where(a => a.ULN == uln && a.AccountLegalEntityId == accountLegalEntityId).ToListAsync();
+
+            switch (apprenticeships.Count)
+            {
+                case 0:
+                    return null;
+                case 1:
+                    return apprenticeships[0].Map(_dbContext.CollectionPeriods.AsEnumerable());
+                default:
+                    throw new InvalidIncentiveException($"Found duplicate ULNs {uln} within account legal entity {accountLegalEntityId}");
```

Comment:
This is a business rule so shouldn't be implemented here.  Just return a list of ApprenticeshipIncentiveModel and let the command handler do the check and thrown the domain exception.